tests/dm_test.py

```diff
@@ -39,6 +39,21 @@ def _clean_up(self):
 
         os.unlink(self.dev_file)
 
+class DevMapperGetSubsystemFromName(DevMapperTestCase):
+    def _destroy_lvm(self):
+        run("vgremove --yes libbd_dm_tests >/dev/null 2>&1")
+        run("pvremove --yes %s >/dev/null 2>&1" % self.loop_dev)
+
+    def test_get_subsystem_from_name(self):
+        """Verify that it is possible to get device subsystem from its name"""
+        run("pvcreate -ff -y %s >/dev/null 2>&1" % self.loop_dev)
+        run("vgcreate -s10M libbd_dm_tests %s >/dev/null 2>&1" % self.loop_dev)
+        run("lvcreate -n subsystem_test -L50M libbd_dm_tests >/dev/null 2>&1")
+        self.addCleanup(self._destroy_lvm)
```

Comment:
Please add the cleanup first, that way it will run even if the creation fails half way.

Reply:
Fixed.